plugin/storage/es/spanstore/reader.go

```diff
@@ -199,9 +200,16 @@ func timeRangeIndices(indexName, indexDateLayout string, startTime time.Time, en
 	var indices []string
 	firstIndex := indexWithDate(indexName, indexDateLayout, startTime)
 	currentIndex := indexWithDate(indexName, indexDateLayout, endTime)
+
+	reduce := -24 * time.Hour
+	if strings.HasSuffix(indexDateLayout, "15") {
```

Comment:
It's a shame `time.stdHour` isn't exported.

Reply:
A cleaner way is to pass freq directly as parameter, not infer it back from format string

Reply:
Yeah, I also thought previously, but if pass freq directly then will also judge this variable too(`if freq == "day"` or move this judgment to `ClientBuilder` of `config.go` ), basically it same as currently.

And another side, other functions will be become more complicated include `NewSpanReader` `SpanReaderParams` `SpanReader` `addRemoteReadClusters` `timeRangeIndices` `createSpanWriter` and reader relevant factory functions and all of unit test functions. and change the `ClientBuilder` of `config.go`.

Actually, I always thought this commit should be small and should not change more code, but if we need, I can do it

Reply:
fair enough. As Albert said, it would be good too at least use a named constant shared between config and span reader

Reply:
ok, I see, now that we decide to add the new constant, how about to separate `jaeger-span` and `jaeger-service` rollover?
because the reason can refer [here](https://github.com/jaegertracing/jaeger/pull/2935#issuecomment-826650749)
I would add two params `index-rollover-span-frequency` and `index-rollover-span-frequency`, do you have any suggestions?

Reply:
Sounds good. I would suggest these names `index-rollover-frequency-spans` and `index-rollover-frequency-services`